test/quaternion_unit_test.js

```diff
@@ -0,0 +1,38 @@
+import * as THREE from 'three';
+import assert from 'assert';
+/* global describe, it */
+
+describe('Quaternion', function () {
+    it('should compare imprecise Quaternion.slerp with Quaternion.slerpFlat with close values', () => {
+        const angleA = Math.PI * 0.3330;
+        const angleB = Math.PI * 0.3329;
+        const A = new THREE.Quaternion().setFromAxisAngle(new THREE.Vector3(0, 0, 1), angleA);
+        const B = new THREE.Quaternion().setFromAxisAngle(new THREE.Vector3(0, 0, 1), angleB);
+
+        const slerp = 0.25;
+        const slerpResult = [];
+        THREE.Quaternion.slerpFlat(slerpResult, 0, A.toArray(), 0, B.toArray(), 0, slerp);
+        const R = new THREE.Quaternion().fromArray(slerpResult);
+        const O = A.clone().slerp(B, slerp);
+
+        const F = new THREE.Euler().setFromQuaternion(R);
+        const S = new THREE.Euler().setFromQuaternion(O);
+
+        const result = (angleA * (1 - slerp)) + (angleB * (slerp));
+        assert.ok(result - F.z < 1e-15);
```

Comment:
These 2 tests are only testing three.js features, so I don't think they should be in iTowns repository.

Reply:
yes exactly, this allows us to test upstream changes and have a signal to make the change to `Quaternion.slerp`, I can add a message to the test

Reply:
I don't understand what signal this will send?

This test will pass, regardless of when upstream change its `Quaternion.slerp` method.
If you plan to use the "fixed" version of `Quaternion.slerp`, you could just wait for the next threejs release - instead of pushing this temp fix.

Reply:
If someone upgrades THREE  to v94, in itowns, it will be indicated that it can also replace `Quaternion.preciseSlerp` by `Quaternion.slerp`,  I can add a message to the test to be more explicit

Reply:
I'll add a comparison with preciseSlerp